stdlib/lang-parser/parserModule/lexer.bal

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/log;
+
+# Lexer, tokenize the lexemes.
+# the characters pulled from the bufferReader is passed through
+# symbol recognizers, identifers, reserved words and numeric recognizers
+# to build tokens
+public type Lexer object {
+    TokenMapper tknMapper = new;
```

Comment:
Is there any specific reason to use `tknMapper` instead of `tokenMapper`? If not, better to rename to improve readability. In general, better to use a meaningful variable name in most of the time.